packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -440,8 +454,28 @@ class ApkManifestData {
     _Element launchActivity;
     for (_Element activity in activities) {
       final _Attribute enabled = activity.firstAttribute('android:enabled');
-      if (enabled == null || enabled.value.contains('0xffffffff')) {
-        launchActivity = activity;
+      final Iterable<_Element> intentFilters =
+          activity.allElements('intent-filter').cast<_Element>();
+      if (enabled!=null && !enabled.value.contains('0xffffffff'))
+        continue;
+      for (_Element element in intentFilters) {
+        final _Element action = element.firstElement('action');
+        final _Element category = element.firstElement('category');
+        final String actionAttributeValue =
+            action?.firstAttribute('android:name')?.value;
+        final String categoryAttributeValue =
+            category?.firstAttribute('android:name')?.value;
+        if ((actionAttributeValue != null &&
```

Comment:
Please break each of these 2 conditions out into local variables that explain what they mean.